web/src/message_events.ts

```diff
@@ -439,9 +467,12 @@ export function update_messages(events) {
         const new_stream_id = event.new_stream_id;
         // old_stream_id will be present and valid for all stream messages.
         const old_stream_id = event.stream_id;
+        // TODO(evy): is this true?
+        assert(old_stream_id !== undefined);
```

Comment:
This fails a Node test, and also there’s an `if (… && old_stream_id && …)` lower down below the `// This logic is a bit awkward` comment, so this needs further scrutiny.

Reply:
> There are a handful of things I made assumptions about that I need to check, one of which is related to a failing test, but opening this as a draft for now in case anyone wants to take a first pass review on it.

(from the PR description)

Reply:
Okay I'm pretty sure `old_stream_id` has to be defined if we're doing a stream message move. Which is the case in the code block you're referring to, because it's in an `else` after `!topic_edited && !stream_changed`